From: Jake Oshins <jakeo@microsoft.com>
Date: Tue, 8 Mar 2016 12:38:19 +0100
Patch-mainline: submitted (lkml - https://www.mail-archive.com/linux-kernel@vger.kernel.org/msg1097519.html)
Subject: hv: Reverse order of resources in hyperv_mmio
References: bnc#965924

A patch later in this series allocates child nodes in this resource
tree. For that to work, this tree needs to be sorted in ascending order.

For SLE12-SP1.

Signed-off-by: Jake Oshins <jakeo@microsoft.com>
Acked-by: <ohering@suse.de>
---
 drivers/hv/vmbus_drv.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/hv/vmbus_drv.c b/drivers/hv/vmbus_drv.c
index 3e56fec..56e3e3f 100644
--- a/drivers/hv/vmbus_drv.c
+++ b/drivers/hv/vmbus_drv.c
@@ -1099,7 +1099,6 @@ static acpi_status vmbus_walk_resources(struct acpi_resource *res, void *ctx)
 	new_res->end = end;
 
 	/*
-	 * Stick ranges from higher in address space at the front of the list.
 	 * If two ranges are adjacent, merge them.
 	 */
 	do {
@@ -1120,7 +1119,7 @@ static acpi_status vmbus_walk_resources(struct acpi_resource *res, void *ctx)
 			break;
 		}
 
-		if ((*old_res)->end < new_res->start) {
+		if ((*old_res)->start > new_res->end) {
 			new_res->sibling = *old_res;
 			if (prev_res)
 				(*prev_res)->sibling = new_res;
